etl/src/main/java/org/pih/hivmigration/etl/sql/EncounterMigrator.groovy

```diff
@@ -122,6 +122,19 @@ class EncounterMigrator extends SqlMigrator {
               and e.encounter_date is not null   # TODO: figure out what to do with these https://pihemr.atlassian.net/browse/UHM-3237
             ;
         ''')
+
+        executeMysql("Log warnings for encounters with null encounter dates", '''
+            INSERT INTO hivmigration_data_warnings (patient_id, encounter_id, field_name, note)
+            SELECT patient_id,
+                   encounter_id,
+                   'encounter_date',
+                   CONCAT('Encounter date is null. Encounter not migrated to encounter table. ',
```

Comment:
+1 for making it clear that the encounter has not been migrated.  This would probably have been my first use case for a "priority" flag, making this an "ERROR", not a warning... no need to add at this point, but might be worth considering as we go forward if we want to explicitly flag issues like these that lead to data not being migrated as higher priority than those that just point out bad data that we still are able to migrate.

Reply:
Actually, none of these encounters have data associated with them, from any of
- hiv_observations
- hiv_ordered_other
- hiv_exam_extra
- hiv_exam_vital_signs
- hiv_dispensing
- hiv_dispensing_meds
- hiv_lab_results

They're probably just crap. We may not even want to warn about them. There is no data.

Reply:
Makes sense @brandones ... @mseaton  thoughts on if there is any other place these encounters might be used?

Agree that if they are totally bogus no need to warn about then...

Reply:
Just pinging @mseaton again to make sure there isn't anything we might be missing where we'd need to warn about this.

Also @brandones  is the date entered for all these encounters way in the past?  (just thinking of the admittedly unlikely possibility that between now and the time of migration an encounter will be added to the system without an encounter datetime but that *does* have some important data we don't want to lose.. so would be good to see if this is an "old" error or something that is stll occurring)

Reply:
Well those aren't the only tables that foreign key to encounter.  And some encounters are valid on their own.  But I haven't done any research into this so really have no idea if these encounters might be valid or not.  Not much downside to logging them as issues.